packages/expo-brightness/ios/BrightnessModule.swift

```diff
@@ -0,0 +1,80 @@
+import ExpoModulesCore
+
+public class BrightnessModule: Module {
+  private var hasListeners = false
+
+  public func definition() -> ModuleDefinition {
+    Name("ExpoBrightness")
+
+    OnCreate {
+      appContext?.permissions?.register([
+        BrightnessPermissionsRequester()
+      ])
+    }
+
+    Events(BrightnessEvents.didChange)
+
+    AsyncFunction("getPermissionsAsync") { (promise: Promise) in
+      guard let permissions = appContext?.permissions else {
+        throw PermissionsModuleNotFoundException()
+      }
+      permissions.getPermissionUsingRequesterClass(
+        BrightnessPermissionsRequester.self,
+        resolve: promise.resolver,
+        reject: promise.legacyRejecter)
+    }
+
+    AsyncFunction("requestPermissionsAsync") { (promise: Promise) in
+      guard let permissions = appContext?.permissions else {
+        throw PermissionsModuleNotFoundException()
+      }
+      permissions.askForPermission(
+        usingRequesterClass: BrightnessPermissionsRequester.self,
+        resolve: promise.resolver,
+        reject: promise.legacyRejecter)
+    }
+
+    // A Double was the only way to satisfy the compiler and avoid a cast. It won't accept a CGFloat argument
+    // "Contextual closure type '() throws -> ()' expects 0 arguments, but 1 was used in closure body"
+    AsyncFunction("setBrightnessAsync") { (brightnessValue: Double) in
+      UIScreen.main.brightness = brightnessValue
+    }.runOnQueue(.main)
+
+    AsyncFunction("getBrightnessAsync") {
+      return UIScreen.main.brightness
+    }
+
+    OnStartObserving {
+      hasListeners = true
+      NotificationCenter.default.addObserver(
+        self,
+        selector: #selector(self.brightnessDidChange),
+        name: UIScreen.brightnessDidChangeNotification,
+        object: nil)
+    }
+
+    OnStopObserving {
+      hasListeners = false
+      NotificationCenter.default.removeObserver(
+        self,
+        name: UIScreen.brightnessDidChangeNotification,
+        object: nil)
+    }
+
+    // Are all of these still needed?
```

Comment:
Unfortunately yes 😞
We're generating JS mocks only based on the iOS platform, here is the generated config with all modules and functions listed: https://github.com/expo/expo/blob/07e8e9756140862e6a753fe1989402ef8a7d4036/packages/jest-expo/src/preset/expoModules.js#L171

I'm looking into how we could improve that (best if can just drop the generated config).
A bit related: I was also thinking about adding additional modifiers, so for example we could write:
```swift
AsyncFunction("getSystemBrightnessAsync")
  .unavailable("This function is not available on iOS platform")

AsyncFunction("deprecatedFunction") { /* do things */ }
  .deprecated("This function has been deprecated in favor of anotherFunction")
```
And then an appropriate error/warning would be logged on the JS side, what do you think?

Reply:
Ah, I see. 
This looks really good, especially for the deprecations 👍